ext/multicore/rpmsg_lite/lib/rpmsg_lite/porting/platform/lpc5411x/rpmsg_platform.c

```diff
@@ -0,0 +1,330 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "rpmsg_platform.h"
+#include "rpmsg_env.h"
+
+#include "fsl_device_registers.h"
+#include "fsl_mailbox.h"
+
+#if defined(RL_USE_MCMGR_IPC_ISR_HANDLER) && (RL_USE_MCMGR_IPC_ISR_HANDLER == 1)
+#include "mcmgr.h"
+#endif
+
+static int isr_counter = 0;
+static int disable_counter = 0;
+static void *lock;
+
+#if defined(RL_USE_MCMGR_IPC_ISR_HANDLER) && (RL_USE_MCMGR_IPC_ISR_HANDLER == 1)
+static void mcmgr_event_handler(uint16_t vring_idx, void *context)
+{
+    env_isr(vring_idx);
+}
+#else
+void MAILBOX_IRQHandler(void)
+{
+    mailbox_cpu_id_t cpu_id;
+#if defined(__CM4_CMSIS_VERSION)
+    cpu_id = kMAILBOX_CM4;
+#else
+    cpu_id = kMAILBOX_CM0Plus;
+#endif
+
+    uint32_t value = MAILBOX_GetValue(MAILBOX, cpu_id);
+
+    if (value & 0x01)
+    {
+        env_isr(0);
+        MAILBOX_ClearValueBits(MAILBOX, cpu_id, 0x01);
+    }
+    if (value & 0x02)
+    {
+        env_isr(1);
+        MAILBOX_ClearValueBits(MAILBOX, cpu_id, 0x02);
+    }
+}
+    /* Add for ARM errata 838869, affects Cortex-M4, Cortex-M4F Store immediate overlapping
+      exception return operation might vector to incorrect interrupt */
+#if defined __CORTEX_M && (__CORTEX_M == 4U)
+    __DSB();
+#endif
+#endif
+
+int platform_init_interrupt(int vq_id, void *isr_data)
+{
+    /* Register ISR to environment layer */
+    env_register_isr(vq_id, isr_data);
+
+    env_lock_mutex(lock);
+
+    assert(0 <= isr_counter);
+    if (!isr_counter)
```

Comment:
I agree. However the files come from RPMsg-Lite project version 1.2.0. So it will be fixed in the next release of RPMsg-Lite and then it can get into Zephyr. I would not change the file in Zephyr now in order for the file content to be the same as in referenced version of RPMsg-Lite.